paasta_tools/check_oom_events.py

```diff
@@ -104,13 +105,19 @@ def read_oom_events_from_scribe(cluster, superregion, num_lines=1000):
         lines=num_lines,
         superregion=superregion,
     )
-    for line in stream:
-        try:
-            j = json.loads(line)
-            if j.get("cluster", "") == cluster:
-                yield j
-        except json.decoder.JSONDecodeError:
+    try:
+        for line in stream:
+            try:
+                j = json.loads(line)
+                if j.get("cluster", "") == cluster:
+                    yield j
+            except json.decoder.JSONDecodeError:
+                pass
+    except StreamTailerSetupError as e:
+        if "No data in stream" in str(e):
             pass
+        else:
+            raise e
```

Comment:
`if "No data in stream" not in str(e):`
&ensp;&ensp;`raise e` ?

Reply:
Eh, I guess I kinda feel like this way is a little easier to reason about?  Having the double-negative there is weird.